core/src/components/modal/modal.tsx

```diff
@@ -237,6 +239,11 @@ export class Modal implements ComponentInterface, OverlayInterface {
    */
   @Event({ eventName: 'ionModalDidDismiss' }) didDismiss!: EventEmitter<OverlayEventDetail>;
 
+  /**
+   * Emitted after the modal breakpoint has changed.
+   */
+  @Event() ionBreakpointDidChange!: EventEmitter<{breakpoint: number}>;
```

Comment:
Do we need to expose an `ModalCustomEvent` interface with the target set to `HTMLIonModalElement`? 

Similar to what infinite scroll and other components do: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/infinite-scroll/readme.md#interfaces